src/differentials.jl

```diff
@@ -291,14 +316,18 @@ function itself, when that function is not a closure.
 const NO_FIELDS = DNE()
 
 """
-    refine_differential(𝒟::Type, der)
+    refine_differential([𝒟::Type, ]der)
 
 Converts, if required, a differential object `der`
 (e.g. a `Number`, `AbstractDifferential`, `Matrix`, etc.),
 to another  differential that is more suited for the domain given by the type 𝒟.
 Often this will behave as the identity function on `der`.
 """
 function refine_differential(::Type{<:Union{<:Real, AbstractArray{<:Real}}}, w::Wirtinger)
+    w = refine_differential(w)
     return wirtinger_primal(w) + wirtinger_conjugate(w)
 end
-refine_differential(::Any, der) = der  # most of the time leave it alone.
+refine_differential(::Any, der) = refine_differential(der)  # most of the time leave it alone.
+
+refine_differential(w::Wirtinger{<:Any,Zero}) = w.primal
```

Comment:
I think better to have multi-arg as target,
and migth need to refine the primal too?
```suggestion
refine_differential(::Any, w::Wirtinger{<:Any,Zero}) = refine_differential(𝒟, w.primal)
```

or even:

Reply:
```suggestion
refine_differential(::Any, w::Wirtinger{<:Any,Zero}) = refine_differential(w.primal)
```

Reply:
After thinking about this a bit, I think having the methods with the domain as their first argument apply only the refinements that actually depend on the domain, and having the single-arg methods handle everything else, makes the most sense, since we can avoid some ambiguities this way.

Reply:
Ok, we can always change things so I don't think either way or the other is a big deal